packages/jest-diff/src/index.ts

```diff
@@ -165,14 +160,15 @@ function compareObjects(
 
 // eslint-disable-next-line no-redeclare
 namespace diff {
-  export type Diff = DiffClass;
   export type DiffOptions = JestDiffOptions;
 }
 
 diff.diffStringsUnified = diffStringsUnified;
+
 diff.diffStringsRaw = diffStringsRaw;
 diff.DIFF_DELETE = DIFF_DELETE;
 diff.DIFF_EQUAL = DIFF_EQUAL;
 diff.DIFF_INSERT = DIFF_INSERT;
+diff.Diff = Diff;
 
 export = diff;
```

Comment:
We can probably make this into ES modules now, might mean the API is easier. Ditch the `export =` stuff

Reply:
In addition to rewriting `import` and `export` statements, is there any configuration change?

Reply:
No, I don't think so

Reply:
Good bye and good riddance to the obscure TypeScript error. Thank you, my friend.